src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/image.py

```diff
@@ -31,24 +33,33 @@ def acr_image_import(cmd,
     slash = source_image.find('/')
 
     if dot < 0 or slash < 0:
-        #todo we might support import images from docker hub
         raise CLIError(INVALID_SOURCE_IMAGE)
 
     source_registry = source_image[:dot]
     if not source_registry:
         raise CLIError(INVALID_SOURCE_IMAGE)
+
     if not resource_id:
-        resource_id = get_resource_id_by_registry_name(cmd.cli_ctx, source_registry)
-    source_image = source_image[slash + 1 :]
+        resource_id = get_resource_id_by_registry_name(cmd.cli_ctx, source_registry, SOURCE_REGISTRY_NOT_FOUND)
+    registry_name_from_resource_id = get_registry_name_by_resource_id(
+        resource_id)
+    if source_registry != registry_name_from_resource_id:
```

Comment:
Not getting this condition. `resource_id` is obtained from `source_registry`, `registry_name_from_resource_id` is obtained from `resource_id`, `shouldn't registry_name_from_resource_id` always match `source_registry`?